### ISSUE TITLE
Account::try_from lifetime is not useable

### ISSUE THREAD

**Issue by juchiast** [coral-xyz/anchor#3341]
Since 0.29, `Account::try_from` signature changed from
```rust
fn try_from(info: &AccountInfo<'a>) -> Result<Account<'a, T>> {
```
to 
```rust
fn try_from(info: &'a AccountInfo<'a>) -> Result<Account<'a, T>> {
```
If I have an owned `AccountInfo<'a>` variable, I cannot borrow it again with lifetime `'a` because the current scope is always smaller, for example:
```rust
fn foo<'a>(info: AccountInfo<'a>) {
    Account::try_from(&info); // not possible because reference cannot live as long as 'a
}
```

**Comment by juchiast** [comment]
My use-case will be resolved with https://github.com/coral-xyz/anchor/pull/3342 

Generally, this is still an issue because it is very hard to create a `&'a AccountInfo<'a>`. `AccountInfo<'a>` invariant and cannot be cast to a shorter lifetime `AccountInfo<'b>`.

**Comment by acheroncrypto** [comment]
> If I have an owned `AccountInfo<'a>` variable, I cannot borrow it again with lifetime `'a` because the current scope is always smaller, for example:
> 
> ```rust
> fn foo<'a>(info: AccountInfo<'a>) {
>     Account::try_from(&info); // not possible because reference cannot live as long as 'a
> }
> ```

Yeah, this should be fixed with https://github.com/coral-xyz/anchor/pull/3340. See the "Details" section of https://github.com/coral-xyz/anchor/pull/2770 for more information about the problem.

**Comment by xenoliss** [comment]
Hello, facing the same issue. Is there any temporary workaround while a proper solution is being implemented? Thanks!

**Comment by Henry-E** [comment]
Guessing this is fixed but not 100% sure so leaving open for now
